post-install/sleep.md

```diff
@@ -0,0 +1,261 @@
+# Fixing Sleep
+
+* Supported version: 0.5.9
+
+So to understand how to fix sleep issues in macOS, we need to first look at what contributes to sleep issues most of the time:
+
+* Incorrectly managed devices(most commonly PCIe based devices)
+
+The reason for this is when devices get an S3 call(or S0 for wake), the driver needs to power down the devices and put into a low state mode(vice versa when waking). Problems arise when such devices don't cooperate with the drivers and the main offenders of these issues are:
+
+* USB Controllers and Devices
+* GPUs
+* Thunderbolt Controllers and Devices
+* NICs(Both Ethernet and Wifi)
+* NVMe Drives
+
+And there are others that can cause sleep issues that aren't directly(or obviously) related to PCI/e:
+ 
+* CPU Power Management
+* Displays
+* NVRAM
+* RTC/System Clocks
+* IRQ Conflicts
+* Audio
+* SMBus
+* TSC
+
+And something many people forget are over and under-clocks:
+ 
+* CPUs
+  * AVX often breaks iGPUs and hurt overall stability
+* Bad RAM(Both overclocks and mismatched RAM)
+  * Even bad/mismatched timings can cause serious issues
+* Factory GPU Overclocks
+  * OEMs commonly push a card a bit too far with their custom VBIOS
+  * Generally these cards will have a physical switch, allowing you to choose a low power VBIOS
+
+## Preparations
+
+
+**In macOS**:
+
+Before we get in too deep, we'll want to first ready our system:
+
+```
+sudo pmset autopoweroff 0 
+sudo pmset powernap 0 
+sudo pmset standby 0
+sudo pmset proximitywake 0
+```
+
+This will do 4 things for us:
+
+1. Disables autopoweroff: This is a form of hibernation
+2. Disables powernap: Used to periodically wake the machine for network, and updates(but not the display)
+3. Disables standby: Used as a time period between sleep and going into hibernation
+4. Disables wake from iPhone/Watch: Specifically when your iPhone or Apple Watch come near, the machine will wake
+
+**In your config.plist**:
+
+While minimal changes are needed, here are the ones we care about:
+
+* `Misc -> Boot -> HibernateMode -> None`
+  * We're gonna avoid the black magic that is S4 for this guide
+* `NVRAM -> Add -> 7C436110-AB2A-4BBB-A880-FE41995C9F82 -> boot-args`
+  * `keepsyms=1`- Makes sure that if a kernel panic does happen during sleep, that we get all the important bits from it
+
+**In your BIOS**:
+
+* Disable:
+  * Wake on LAN
+  * Trusted Platform Module
```

Comment:
Caution: Users that have a TPM installed will run into issues with BitLocker. Windows requires the use of a TPM for this feature to work. If users disable this without thinking ahead, their keys to the encryption will be lost. This could spell disaster for for getting data back. I need to find where else this is detailed in the guides for those that BitLocker on. There should be a warning to turn BitLocker off before hand. Just in case.

Reply:
Ah thank you, yeah it should be clarified now. None of the other guides mention TPM afaik, at least couldn't find them mentioned myself